### PR TITLE
[serve] Add more verbose logging in the Serve controller and REST API

### LINKED ISSUE #33203: [serve] Add controller logs for when replicas finish starting up
Currently trying to help a user debug an issue and we have logs for when replicas are started but not when they finish initializing. This is important context.

For example we have:
```
 155 2023-03-10T11:14:57-06:00 INFO 2023-03-10 09:14:57,180 controller 979 deployment_state.py:625 - Actor for replica D#DYFbUC crashed, marking it unhealthy immediately.
 154 2023-03-10T11:14:57-06:00 WARNING 2023-03-10 09:14:57,187 controller 979 deployment_state.py:1506 - Replica D#DYFbUC of deployment D failed health check, stopping it.
 153 2023-03-10T11:14:57-06:00 INFO 2023-03-10 09:14:57,327 controller 979 deployment_state.py:1310 - Adding 1 replica to deployment 'D'.
```

Would be good to also have:
```
 153 2023-03-10T11:14:57-06:00 INFO 2023-03-10 09:14:57,327 controller 979 deployment_state.py:1310 - Starting replica <REPLICA_ID> for deployment 'D'.
 153 2023-03-10T11:14:57-06:00 INFO 2023-03-10 09:14:57,327 controller 979 deployment_state.py:1310 - Replica <REPLICA_ID> for deployment 'D' finished initializing after <TIME_S>.
```

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Improve logging to help diagnose issues.

## Related issue number

Closes https://github.com/ray-project/ray/issues/33203

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
Only concern here is the output on the terminal getting too spammy, but I think all of these messages are useful.

File: python/ray/serve/_private/deployment_state.py [hunk 1]
```diff
@@ -290,7 +290,7 @@ def start(self, deployment_info: DeploymentInfo, version: DeploymentVersion):
             deployment_info.deployment_config.is_cross_language
         )
 
-        logger.debug(
+        logger.info(
```
Comment (sihanwang41) [inline python/ray/serve/_private/deployment_state.py:293]: nit: For consistency to show replica operation, also add corresponding info in the grace_stop() func

**Review by sihanwang41 — APPROVED** [review]

File: python/ray/serve/_private/deployment_state.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/_private/deployment_state.py:293]: done

Comment (edoakes) [reply python/ray/serve/_private/deployment_state.py:293]: done

**Comment by edoakes** [comment]
I followed the instructions to fix the DCO and it completely borked this PR. Will fix and re-open...

**Pull request closed without merging**